presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorTableMetadata.java

```diff
@@ -22,11 +24,14 @@
 import static java.util.Collections.emptyMap;
 import static java.util.Objects.requireNonNull;
 
+@Immutable
 public class ConnectorTableMetadata
 {
     private final SchemaTableName table;
     private final Optional<String> comment;
+    @SuppressWarnings("Immutable")
     private final List<ColumnMetadata> columns;
+    @SuppressWarnings("Immutable")
     private final Map<String, Object> properties;
```

Comment:
We should consider to introduce guava dep to presto-spi module so we can use immutable collections instead.

Reply:
That can't happen since Guava is a common source of dependency conflicts

Reply:
Error Prone should be able to analyze the bytecode and see that the assignments are creating immutable copies.

Reply:
According to [ImmutableAnalysis](https://github.com/google/error-prone/blob/v2.4.0/core/src/main/java/com/google/errorprone/bugpatterns/threadsafety/ImmutableAnalysis.java#L293-L311), I guessed error prone only checks the field's type, not how it was created.